File: easy/haskell/onboarding.hs
import Control.Monad (forever)

main :: IO ()
main = forever $ do
    enemy1 <- getLine
    dist1 <- readLn
    enemy2 <- getLine
    dist2 <- readLn

    if dist1 < dist2 then
        putStrLn enemy1
    else
        putStrLn enemy2

File: easy/c/onboarding.c
#include <stdio.h>

int main() {
    char enemy1[50], enemy2[50];
    int dist1, dist2;

    while (1) {
        scanf("%s %d", enemy1, &dist1);
        scanf("%s %d", enemy2, &dist2);

        if (dist1 < dist2) {
            printf("%s\n", enemy1);
        } else {
            printf("%s\n", enemy2);
        }
    }

    return 0;
}
